advanced/01-pihole.conf

```diff
@@ -35,7 +35,7 @@ server=@DNS2@
 
 interface=@INT@
 
-cache-size=10000
```

Comment:
This change is unnecessary.  The end result can be accomplished less code if you leave the default value in place and then use the following to replace only the value if it changed.  The following would go in the basic-install.sh file.
`sed -i 's/\(cache-size=\).*/\1${CACHE_SIZE}/' ${dnsmasq_pihole_01_location}`

Reply:
I disagree.

Reply:
If the default value is left in place then the docker image will always have that value. Removing the configuration is beneficial. @diginc FYI.

Reply:
The default value would be replaced with the following in the basic-install.sh.  I fixed a typo from my earlier comment.  Bash will not expand variables inside single quotes.  Just by searching for the existing variable name you can replace the value.
`sed -i "s/\(cache-size=\).*/\1${CACHE_SIZE}/" ${dnsmasq_pihole_01_location}`

Reply:
If the docker image creation used basic-install then there'd be a point. It doesn't however.